tests/serac_component_bc.cpp

```diff
@@ -9,49 +9,67 @@
+  serac::mesh::InputInfo::defineInputFileSchema(mesh_table);
+
+  // Physics
+  auto& solid_solver_table = inlet.addTable("nonlinear_solid", "Finite deformation solid mechanics module");
+  serac::NonlinearSolid::InputInfo::defineInputFileSchema(solid_solver_table);
+
+  // Verify input file
+  if (!inlet.verify()) {
+    SLIC_ERROR("Input file failed to verify.");
+  }
+}
+
 TEST(component_bc, qs_solve)
 {
   MPI_Barrier(MPI_COMM_WORLD);
 
-  // Open the mesh
-  std::string mesh_file = std::string(SERAC_REPO_DIR) + "/data/meshes/square.mesh";
+  std::string input_file_path =
+      std::string(SERAC_REPO_DIR) + "/data/input_files/tests/nonlinear_solid/serac_component_bc/qs_solve.lua";
 
-  auto pmesh = buildMeshFromFile(mesh_file, 2, 0);
+  // Create DataStore
+  axom::sidre::DataStore datastore;
 
-  int dim = pmesh->Dimension();
+  // Initialize Inlet and read input file
+  auto inlet = serac::input::initialize(datastore, input_file_path);
 
-  // Set the linear solver params
-  IterativeSolverParameters params = {.rel_tol     = 1.0e-6,
-                                      .abs_tol     = 1.0e-8,
-                                      .print_level = 0,
-                                      .max_iter    = 5000,
-                                      .lin_solver  = LinearSolver::MINRES,
-                                      .prec        = HypreSmootherPrec{mfem::HypreSmoother::l1Jacobi}};
-  params.rel_tol                   = 1.0e-8;
-  params.abs_tol                   = 1.0e-12;
+  defineInputFileSchema(inlet);
 
-  NonlinearSolverParameters nl_params = {.rel_tol = 1.0e-3, .abs_tol = 1.0e-6, .max_iter = 5000, .print_level = 1};
-  nl_params.rel_tol                   = 1.0e-6;
-  nl_params.abs_tol                   = 1.0e-8;
+  // Build the mesh
+  auto mesh_info      = inlet["main_mesh"].get<serac::mesh::InputInfo>();
+  auto full_mesh_path = serac::input::findMeshFilePath(mesh_info.relative_mesh_file_name, input_file_path);
+  auto mesh           = serac::buildMeshFromFile(full_mesh_path, mesh_info.ser_ref_levels, mesh_info.par_ref_levels);
 
-  // Define the solver object
-  NonlinearSolid solid_solver(1, pmesh, {params, nl_params});
+  // Define the solid solver object
+  auto                  solid_solver_info = inlet["nonlinear_solid"].get<serac::NonlinearSolid::InputInfo>();
+  serac::NonlinearSolid solid_solver(mesh, solid_solver_info);
 
-  // boundary attribute 1 (index 0) is fixed (Dirichlet) in the x direction
-  std::set<int> ess_bdr = {1};
+  int dim = mesh->Dimension();
 
   // define the displacement vector
   auto disp_coef = std::make_shared<StdFunctionCoefficient>([](mfem::Vector& x) { return x[0] * -1.0e-1; });
 
   // Pass the BC information to the solver object setting only the z direction
-  solid_solver.setDisplacementBCs(ess_bdr, disp_coef, 0);
+  solid_solver.setDisplacementBCs(solid_solver_info.boundary_conditions.at("displacement").attrs, disp_coef, 0);
```

Comment:
While the `at` lookup instead of a loop is nice, I think it adds unnecessary complexity and duplication, especially since this will change in the near future anyway.